x/stakeibc/keeper/icqcallbacks_withdrawal_reward_balance_test.go

```diff
@@ -0,0 +1,231 @@
+package keeper_test
+
+import (
+	"fmt"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/gogoproto/proto"
+
+	epochtypes "github.com/Stride-Labs/stride/v16/x/epochs/types"
+	icqtypes "github.com/Stride-Labs/stride/v16/x/interchainquery/types"
+	"github.com/Stride-Labs/stride/v16/x/stakeibc/keeper"
+	"github.com/Stride-Labs/stride/v16/x/stakeibc/types"
+)
+
+// WithdrawalRewardBalanceCallback will trigger TransferRewardTokensHostToTrade
+// Therefore we need to setup traderoute fields used in the entire transfer (with pfm)
+func (s *KeeperTestSuite) SetupWithdrawalRewardBalanceCallbackTestCase() BalanceQueryCallbackTestCase {
+	// Create the connection between Stride and HostChain with the withdrawal account initialized
+	withdrawalAccountOwner := fmt.Sprintf("%s.%s", HostChainId, types.ICAAccountType_WITHDRAWAL.String())
+	withdrawalChannelId, withdrawalPortId := s.CreateICAChannel(withdrawalAccountOwner)
+	withdrawalAddress := s.IcaAddresses[withdrawalAccountOwner]	
+	withdrawalConnectionId, _, _ := s.App.StakeibcKeeper.IBCKeeper.ChannelKeeper.
+		GetChannelConnection(s.Ctx, withdrawalPortId, withdrawalChannelId)	
+
+	hostICA := types.ICAAccount{
+		ChainId: HostChainId,
+		Type: types.ICAAccountType_WITHDRAWAL,
+		ConnectionId: withdrawalConnectionId,
+		Address: withdrawalAddress,
+	}
+	rewardICA := types.ICAAccount{
+		ChainId: "noble-01",
+		Type: types.ICAAccountType_CONVERTER_UNWIND,
+		Address: HostICAAddress, // doesn't have to exist or be connected, but must be betch32 decode-able
+	}
+	tradeICA := types.ICAAccount{
+		ChainId: OsmoChainId,
+		Type: types.ICAAccountType_CONVERTER_TRADE,
+		Address: HostICAAddress, // doesn't have to exist or be connected, but must be betch32 decode-able		
+	}
+
+	// Must initialize these or they will serialize differently from the default values
```

Comment:
I was using the traderoute as the type which gets marshalled and stored as callback data.  In that case an unintialized INT or DEC type on these fields will have literally random data in the memory location because it is not a basic type so go does not auto init but it does make room for the expected type, and storing it down and reading it back will cause a mismatch in tests that check the callback data was able to be read correctly